tests/densify/test_interface_patchmatchnet.py

```diff
@@ -0,0 +1,160 @@
+"""Unit tests for the interface for PatchmatchNet
+
+   _C1_ (0, 0, 1)
+ /     \\
+|       |
+|       |C0 (1, 0, 0)                      ---------P2(0.5, 0, 0.5)-----  camera plane
+ \\_ _ /
+  C2 (-0.5, 0, -0.5 * sqrt(3))                 P1(-0.1, -1.0, 0.1)
+
+In this unit test, we make a simple scenario that three cameras {C0, C1, C2} are around a circle (radius = 1.0).
+Every camera's pose is towards the center of the circle (0, 0, 0).
+Two track points {P1, P2} are set on the camera plane and under the camera plane.
+
+Authors: Ren Liu
+"""
+import unittest
+
+import numpy as np
+from gtsam import PinholeCameraCal3Bundler, Cal3Bundler, Pose3, Rot3, Point3
+
+from gtsfm.common.image import Image
+from gtsfm.common.gtsfm_data import GtsfmData, SfmTrack
+from gtsfm.densify.patchmatchnet_data import PatchmatchNetData, MIN_DEPTH_PERCENTILE, MAX_DEPTH_PERCENTILE
+
+# set the default image size as 800x600, with 3 channels
+DEFAULT_IMAGE_W = 800
+DEFAULT_IMAGE_H = 600
+DEFAULT_IMAGE_C = 3
+
+# set default track points, the coordinates are in the world frame
+DEFAULT_TRACK_POINTS = [Point3(0.5, 0.5, 0), Point3(-0.1, -1.0, 0.1)]
+DEFAULT_NUM_TRACKS = len(DEFAULT_TRACK_POINTS)
+
+# set default camera intrinsics
+DEFAULT_CAMERA_INTRINSICS = Cal3Bundler(
+    fx=100.0,
+    k1=0,
+    k2=0,
+    u0=DEFAULT_IMAGE_W // 2,
+    v0=DEFAULT_IMAGE_H // 2,
+)
+# set default camera poses as described above
+DEFAULT_CAMERA_POSES = [
+    Pose3(Rot3(np.array([[0, 0, -1], [0, 1, 0], [1, 0, 0]])), np.array([1, 0, 0])),
+    Pose3(Rot3(np.array([[-1, 0, 0], [0, 1, 0], [0, 0, -1]])), np.array([0, 0, 1])),
+    Pose3(
+        Rot3(np.array([[0.5 * np.sqrt(3), 0.0, 0.5], [0, 1, 0], [-0.5, 0, 0.5 * np.sqrt(3)]])),
```

Comment:
the sqrt(3) one is a bit hard to understand intuitively.

I would recommend using Lookat from here:
https://github.com/borglab/gtsam/blob/develop/python/gtsam/examples/SFMdata.py

and setting up the circle poses analytically, as shown here:
https://github.com/borglab/gtsam/blob/develop/python/gtsam/utils/circlePose3.py

Reply:
you can see what those poses look like with:
```python
    from gtsfm.utils.viz import plot_poses_3d, set_axes_equal
    import matplotlib.pyplot as plt

    from gtsam.SFMdata import createPoses
    K = Cal3_S2( blah )
    poses = createPoses(K)

    fig = plt.figure()
    ax = fig.gca(projection="3d")

    plot_poses_3d(poses, ax, center_marker_color="k")

    set_axes_equal(ax)
    plt.show()
```

Reply:
<img width="452" alt="Screen Shot 2021-08-21 at 11 03 37 PM" src="https://user-images.githubusercontent.com/16724970/130443829-c2306505-b2f4-47fa-9eef-d209eea90e72.png">